Client/mods/deathmatch/logic/CClientIMGManager.h

```diff
@@ -0,0 +1,44 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto v1.0
+ *               (Shared logic for modifications)
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        mods/shared_logic/CClientIMGManager.h
+ *  PURPOSE:     .img container manager class
+ *
+ *****************************************************************************/
+
+class CClientIMGManager;
+
+#pragma once
+
+#include <list>
+#include "CClientIMG.h"
+
+class CClientIMGManager
+{
+    friend class CClientIMG;
+
+public:
+    CClientIMGManager(class CClientManager* pManager);
+    ~CClientIMGManager();
+
+    void RemoveAll();
+    bool Exists(CClientIMG* pIMG);
+
+    CClientIMG* GetElementFromArchiveID(unsigned char ucStreamID);
+    CClientIMG* GetElementThatLinked(unsigned int uiModel);
+
+    bool        RestoreModel(unsigned int uiModel);
+    static bool IsLinkableModel(unsigned int uiModel);
+
+    std::list<CClientIMG*>::const_iterator IterBegin() { return m_List.begin(); }
+    std::list<CClientIMG*>::const_iterator IterEnd() { return m_List.end(); }
+
+private:
+    void AddToList(CClientIMG* pIMG) { m_List.push_back(pIMG); }
+    void RemoveFromList(CClientIMG* pIMG);
+
+    std::list<CClientIMG*> m_List;
```

Comment:
Use `std::vector<CClientIMG*>`.
Using a `list` for pointers is a really, really, really bad idea.